test/utils/project_holder.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+	boskosclient "sigs.k8s.io/boskos/client"
+	"sigs.k8s.io/boskos/common"
+)
+
+type ProjectHolder struct {
+	c *boskosclient.Client
+	// startCleanup channel signals when we need to stop refresh the resource and release the boskos project.
+	startCleanup chan bool
+	// finishCleanup channel signals when the clean up is finished.
+	finishCleanup chan bool
+}
+
+func NewProjectHolder() (*ProjectHolder, error) {
+	jobName := os.Getenv("JOB_NAME")
+	if jobName == "" {
+		return nil, fmt.Errorf("JOB_NAME required but not provided")
+	}
+	c, err := boskosclient.NewClient(jobName, "http://boskos", "", "")
+	if err != nil {
+		return nil, err
+	}
+	return &ProjectHolder{
+		c:             c,
+		startCleanup:  make(chan bool),
+		finishCleanup: make(chan bool),
+	}, nil
+}
+
+// AcquireOrDie tries to get a boskso project and spawn a goroutine to refresh it,
+// or fail to get a project and terminate.
+func (ph *ProjectHolder) AcquireOrDie(resourceType string) string {
+	// Try to get a Boskos project
+	klog.Infof("Running in Prow, getting project resourceType = %q", resourceType)
+
+	p := ph.getBoskosProjectOrDie(resourceType)
+	project := p.Name
+	go ph.refresh(project)
+
+	return project
+}
+
+// Release stops the refresh goroutine, and release the boskso project.
+func (ph *ProjectHolder) Release(project string) {
```

Comment:
Does it make sense to have this take in the project or to cache it in the struct?

e.g. 

add to the struct

  project string

then AcquireOrDie does

  ph.project = p.Name

And Release() does not need a parameter